Initial commit of web UI content (CSE-26)

jeremy-moffitt opened the pull request:
Includes the initial set of web content for the pint web UI. 

- loads live data from https://susepubliccloudinfo.suse.com/v1/
- is styled to conform to www.suse.com look & feel (initial review with UX found the styling acceptable)
- defaults to Microsoft and "all states" for the initial data load
- data is cached for 24 hours in local storage in the browser
- CSP and state selections change the request to susepubliccloudinfo.suse.com , environments and search continue to be handled client-side
- error banners are present for if the data fetch has an error
- load indicator added when the table is updating, serves 2 purposes: users know a load is on-going, and the DOM updates faster when the table is not visible during the update

jeremy-moffitt commented:
It's unclear to me if we'll need to structure this differently to align with hosting options (presumably either an S3 bucket or hosted directly from github).... happy to make changes accordingly.

rjschwei commented on index.html:
```diff
@@ -0,0 +1,422 @@
+  var regions = {};
+  var region = '';
+  var states = {};
+  var current_csp = 'microsoft';//default provider loaded on page load
```
I think we should go in alpha order, otherwise we'll be accused of playing favorite. So alibaba should be first and the default.

It would be nice if this were stored in a cookie and restored from there when the user comes back.

jeremy-moffitt replied:
I'll work on the cookie next. Since alibaba isnt live yet at https://susepubliccloudinfo.suse.com/v1/providers.json  , I'm a little hesitant to load Amazon as the default because it has the longest load time

rjschwei replied:
Alibaba will go live within 2 hours after our gitlab server is back operational. Meaning by the time we get this code into S3 and get it live alibaba will be there.

rjschwei replied:
alibaba data is live :D 

```
-> pint alibaba images --active --region us-east-1
<?xml version='1.0' encoding='UTF-8'?>
<images>
  <image name="sles-15-sp2-chost-byos-v20200831" state="active" replacementname="" publishedon="20200831" deprecatedon="" region="us-east-1" id="m-0xi8akt9fyjoma2l2ro0" deletedon=""/>
</images>
```

jeremy-moffitt replied:
Alibaba is live, and the setting is now preserved in a cookie

jeremy-moffitt replied:
and I should reload the conversation page before typing responses :)

jeremy-moffitt commented on index.html:
```diff
@@ -0,0 +1,422 @@
+						<br>
+						<h1>PINT - The Public Cloud INformation Tracker</h1>
+						<p>This service provides data about the images SUSE publishes in various
+						public cloud frameworks (Amazon, Google, Microsoft and Oracle) as well
```
I will fix this, also highlights that the spacing is weird on this... I think vscode has a spaces vs tabs issue I havent quite worked out

jeremy-moffitt replied:
fixed in the latest revision

bear454 replied:
You're going to be chasing this text forever. Consider an alternative that doesn't list the provider names, or generates the list from the metadata API endpoints.

jeremy-moffitt replied:
Thats a good point, we also don't yet support the servers in the UI (its on the short list of enhancements to do next), will clean this up.

rjschwei commented on README.md:
```diff
@@ -0,0 +1,44 @@
+---------------
+
+The UI can be run on any system that has a basic web server. For development
+purposes using the python3 http server is a option.
```
s/a/an/

jeremy-moffitt replied:
fixed, thanks for pointing that out!

rjschwei reviewed (approved):
LGTM, stumbled over 1 typo

bear454 commented on assets/css/eos-imports.css:
```diff
@@ -0,0 +1,363 @@
+* https://suse.eosdesignsystem.com/
+*/
+
+@import url("https://fonts.googleapis.com/css?family=Work+Sans:300,400,600");
```
Consider including the blobs and serving directly, instead of relying on 3rd party content.

jeremy-moffitt replied:
I had the fonts checked in locally for one version of this, but I waffled on that when I saw the eos design system page recommends using an import https://suse.eosdesignsystem.com/typography

bear454 commented on assets/css/pintui.css:
```diff
@@ -0,0 +1,106 @@
+*/
+
+body.t-work {
+    font-family: 'work',Arial,sans-serif;
```
Verdana should be used as secondary font, not Arial. https://brand.suse.com/brand-system/typography#secondary-fonts

jeremy-moffitt replied:
fixed in latest version, thanks for pointing this out

bear454 reviewed (commented):
Since we're using eos-icons, can we use the navigation arrows instead of static graphics for the sort icons?

https://icons.eosdesignsystem.com/cheatsheet?iconName=arrow_drop_up
https://icons.eosdesignsystem.com/cheatsheet?iconName=arrow_drop_down

bear454 commented:
Open source projects should not include SUSE-specific branding. The SUSE logo should _not_ be distributed here.

jeremy-moffitt commented:
> Since we're using eos-icons, can we use the navigation arrows instead of static graphics for the sort icons?
> 
> https://icons.eosdesignsystem.com/cheatsheet?iconName=arrow_drop_up
> https://icons.eosdesignsystem.com/cheatsheet?iconName=arrow_drop_down

I changed this in the latest version, though its a bit clunky because typically material icons are done using an icon font and since we're importing the table widget wholesale, I downloaded the icons and replaced the ones at the well-known paths the table is looking for. I dont really have a visual preference either way... I didn't see a great replacement for the up/down combo button indicating that the headers can be sorted, and when I tried creating one out of the material icons its much too dark and sorta takes over the header, so I left that one alone for now. 


> Open source projects should not include SUSE-specific branding. The SUSE logo should _not_ be distributed here.

I can remove the logo while keeping the path to it (presumably to be reinserted wherever its ultimately hosted out of) ... but how far do we want to extend this? It's intended as a suse.com subdomain... do we need to pull the description text and footer links too? I definitely need some guidance here... maybe this is better suited to a private repo?

bear454 commented:
> > Open source projects should not include SUSE-specific branding. The SUSE logo should _not_ be distributed here.
> 
> I can remove the logo while keeping the path to it (presumably to be reinserted wherever its ultimately hosted out of) ... but how far do we want to extend this? It's intended as a suse.com subdomain... do we need to pull the description text and footer links too? I definitely need some guidance here... maybe this is better suited to a private repo?

Well, yeah, that needs to be decided. If it's open source and public, it should conform to EOS, if themed at all. If it's going to be specifically SUSE-branded, it needs to be private. Consider the case of another organizing forking it and hosting it with their own _pint_ server. Do we want SUSE branding on that? Absolutely not!

bear454 commented:
Generally, the open source projects aren't themed, or themed neutrally (e.g. EOS), and a SUSE-specific theme is applied in a private fork, or IBS overlay.

jeremy-moffitt commented:
> Well, yeah, that needs to be decided. If it's open source and public, it should conform to EOS, if themed at all. If it's going to be specifically SUSE-branded, it needs to be private. Consider the case of another organizing forking it and hosting it with their own _pint_ server. Do we want SUSE branding on that? Absolutely not!

Do we know who should make this decision? My understanding of the use case boils down to "make this available at pint.suse.com" as a static webpage, rather than "make it a portable web app that anyone could stand up". If thats correct, then we'd either make the repo private on github or keep it internal in gitlab.

jeremy-moffitt commented:
I talked to Ron about this during a call on another topic and his take was that he wants it on suse.com , and if that means it lives somewhere private, thats ok with him. I'll send out email on the topic with a proposal to keep it in gitlab but move it under the team space from the individual space that its in now.

rjschwei commented:
> Generally, the open source projects aren't themed, or themed neutrally (e.g. EOS), and a SUSE-specific theme is applied in a private fork, or IBS overlay.

Fair to a certain extend. On the other hand I could argue that whomever wants to use the server, the cli, and the ui of PINT can have their fork. We have a similar issue with he client cli, just read the man page and it has SUSE plastered all over it. To a certain extend this is somewhat of a philosophical conversation. My question would be, is there an easy way where we can stick the SUSE specific stuff into a certain location and then someone else can do the same for their $FAVORITE and have that theme picked up by setting a variable somewhere?

jeremy-moffitt commented:
> > Generally, the open source projects aren't themed, or themed neutrally (e.g. EOS), and a SUSE-specific theme is applied in a private fork, or IBS overlay.
> 
> Fair to a certain extend. On the other hand I could argue that whomever wants to use the server, the cli, and the ui of PINT can have their fork. We have a similar issue with he client cli, just read the man page and it has SUSE plastered all over it. To a certain extend this is somewhat of a philosophical conversation. My question would be, is there an easy way where we can stick the SUSE specific stuff into a certain location and then someone else can do the same for their $FAVORITE and have that theme picked up by setting a variable somewhere?

I've got a local version that loads the image url and favicon based on settings in a json file thats included in the repo... it loads both "remotely" but we're already doing that with a few other assets. That would allow me to remove the images from the repo, but keep references to them (served up from suse.com). I'm not sure that satisfies the criteria completely or not.

jeremy-moffitt commented:
pushed branding tweaks in a separate commit, will await feedback on if this is better or just different.

jeremy-moffitt commented:
@bear454  please let me know if the adjustments for the branding alleviate your concerns or if I should pursue an alternative such as hosting this privately.

bear454 commented:
> @bear454 please let me know if the adjustments for the branding alleviate your concerns or if I should pursue an alternative such as hosting this privately.

No, it doesn't. Anybody, literally, anybody, could still fork this and by defualt it will present as a SUSE offering. Red Hat, Ubuntu, the government of Iran. Anybody. The `theme.json` file is a good move; we can just patch that into our package instead of including that in the source repo. Better to just describe it in docs (e.g. "you need to create a theme.json file like this...") or provide a sample (e.g. "copy theme.json.sample to theme.json and add your resources...")

jeremy-moffitt commented:
> > @bear454 please let me know if the adjustments for the branding alleviate your concerns or if I should pursue an alternative such as hosting this privately.
> 
> No, it doesn't. Anybody, literally, anybody, could still fork this and by defualt it will present as a SUSE offering. Red Hat, Ubuntu, the government of Iran. Anybody. The `theme.json` file is a good move; we can just patch that into our package instead of including that in the source repo. Better to just describe it in docs (e.g. "you need to create a theme.json file like this...") or provide a sample (e.g. "copy theme.json.sample to theme.json and add your resources...")

In that case I think it makes sense to host it privately and just serve it up as a content from a single source rather than having multiple repos to serve up a relatively simple single page web page. I'll see if we can get that created, and then close this once the content is started over there.

jeremy-moffitt commented:
hosted this privately for now. Perhaps in the future will revisit having a debranded version living here that we sync both directions for updates
